### PR TITLE
Refactor ConnectorIdentity

### PR DESCRIPTION
**Pull request by luohao**
@dain @arhimondr

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java [hunk 1]
```diff
@@ -14,7 +14,6 @@
 package io.prestosql.spi.security;
 
 import java.security.Principal;
-import java.util.Objects;
```
Comment (kokosing) [inline presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java]: please squash this with the previous commit that removed the usage of this import

File: presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java [hunk 2]
```diff
@@ -47,32 +47,13 @@ public String getUser()
         return role;
     }
 
-    @Override
-    public boolean equals(Object o)
```
Comment (kokosing) [inline presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java]: You could mention in the commit what was refactored. Like:
```
Refactor ConnectorIdentity

- Remove not needed and incorrectly implemented hashCode and equals methods
- Return information about role from toString method.
```

Comment (findepi) [reply presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java]: `ConnectorIdentity` equality might be used as Map keys. In fact, it _is_ used in our internal code.
Let's talk. @luohao @kokosing

File: presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java [hunk 3]
```diff
@@ -47,32 +47,13 @@ public String getUser()
...
-        return Objects.hash(user, principal);
-    }
-
     @Override
     public String toString()
     {
         StringBuilder sb = new StringBuilder("ConnectorIdentity{");
         sb.append("user='").append(user).append('\'');
         principal.ifPresent(principal -> sb.append(", principal=").append(principal));
+        role.ifPresent(role -> sb.append(", role=").append(role));
```
Comment (findepi) [inline presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java]: Separate this commit into:
- `Remove ConnectorIdentity#equals`
- `Improve ConnectorIdentity#toString`

`toString` is a minor change, but change to `equals` is not and should not be hidden in a innocently-looking "refactor ..." commit

File: presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java (hunk 2 above)
Comment (kokosing) [reply presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java]: I would merge it as it is, if it is related to our internal code we could handle that there.

**Comment by findepi** [comment]
Just a thought -- i think it would be safer not just to remove `ConnectorIdentity#equals`. You could leave the method with as this:
```
@Override
public boolean equals(Object o)
{
    // TODO remove equals completely after a few months
    throw new UnsupportedOperationException();
}
```

**Comment by luohao** [comment]
> Just a thought -- i think it would be safer not just to remove `ConnectorIdentity#equals`. You could leave the method with as this:
> 
> ```
> @Override
> public boolean equals(Object o)
> {
>     // TODO remove equals completely after a few months
>     throw new UnsupportedOperationException();
> }
> 
> @Override
> public int hashCode()
> {
>     // TODO remove equals completely after a few months
>     // hashCode() needs to be overridden too, since j.u.HashMap may skip equals() check when `==`.
>     throw new UnsupportedOperationException();
> }
> ```

This sounds like a better strategy. I updated the diff and split it into two commits.

**Comment by luohao** [comment]
@findepi Updated the commit message.

**Pull request merged**